### ISSUE TITLE
Error "A property without annotation isn't considered during up-to-date checking." with Gradle 7 and App Distribution Gradle Plugin

### ISSUE THREAD

**Issue by davidbilik** [firebase/firebase-android-sdk#2580]
### [REQUIRED] Step 2: Describe your environment

  * Android Studio version: Artic Fox Canary 14
  * Firebase Component: App Distribution Gradle Plugin
  * Component version:  2.1.0

#### Steps to reproduce:

I am using Android Gradle Plugin version `7.0.0-alpha14` and Gradle version `7.0` and when I try to upload the app to app distribution through the Gradle plugin version `2.1.0` it fails on a task `app:appDistributionUploadDebug` with an error
```
> Task :app:appDistributionUploadDebug FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Some problems were found with the configuration of task ':app:appDistributionUploadDebug' (type 'UploadDistributionTask').
  - Type 'UploadDistributionTask' property 'COMMAND_LINE_PREFIX' is missing an input or output annotation.
    
    Reason: A property without annotation isn't considered during up-to-date checking.
    
    Possible solutions:
      1. Add an input or output annotation.
      2. Mark it as @Internal.
    
    Please refer to https://docs.gradle.org/7.0/userguide/validation_problems.html#missing_annotation for more details about this problem.
  - Type 'UploadDistributionTask' property 'appDistributionProperties' is missing an input or output annotation.
    
    Reason: A property without annotation isn't considered during up-to-date checking.
    
    Possible solutions:
      1. Add an input or output annotation.
      2. Mark it as @Internal.
    
    Please refer to https://docs.gradle.org/7.0/userguide/validation_problems.html#missing_annotation for more details about this problem.
    ```

**Comment by aguatno** [comment]
Hi @davidbilik thanks for sharing this report with us.

Per checking, it appears this is a known issue and there's an internal bug report filed (bugID#183607355) for this issue with gradle, which our App Distribution team is currently evaluating and working on. We don't have any other specifics or exact timeline that we can provide at the moment. I'll let you know if there's any.

Also, we don't usually recommend using unofficial or release candidate builds as they may result to unexpected behaviors or other incompatibility issues. For the meantime, using the official build is recommended.
 
Thanks

**Comment by davidbilik** [comment]
Hey @aguatno thanks for your response. While I understand that using unstable versions is not recommended, Android gradle plugin is not the problem here, it's the gradle 7.0. It throws the same error with the stable AGP 4.1.3 :)

**Comment by leinardi** [comment]
Can confirm: we are getting this error with AGP 4.1.3:

```
FAILURE: Build failed with an exception.
 
* What went wrong:
A problem was found with the configuration of task ':apps:dsa:appDistributionUploadAustriaRelease' (type 'UploadDistributionTask').
  - Type 'UploadDistributionTask' property 'appDistributionProperties' is missing an input or output annotation.
 
    Reason: A property without annotation isn't considered during up-to-date checking.
 
    Possible solutions:
      1. Add an input or output annotation.
      2. Mark it as @Internal.
 
    Please refer to https://docs.gradle.org/7.0/userguide/validation_problems.html#missing_annotation for more details about this problem.
 
* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
```

**Comment by eygraber** [comment]
Gradle 7 is stable. This is blocking from using it.

**Comment by HoodVinci** [comment]
I confirm , I am also reproducing issue with gradle 7.0  and version 2.1.0

**Comment by Aidanvii7** [comment]
Seeing the same issue, this is a major blocker.

Here's the exact configuration I had that didn't work:
Android Studio Arctic Fox Canary 14
 - requires AGP 7.0.0-alpha14
     - requires at least gradle-7.0-rc-1 

Rolling back to Arctic Fox Canary 13 allowed me to use AGP 7.0.0-alpha13, which doesn't require gradle-7.0-rc-1 (using gradle-6.8.2). This config works.

**Comment by tagboola** [comment]
We've confirmed that there is a bug with the App Distribution Gradle plugin when using Gradle 7. We're currently working on a fix. In the meantime, the only workaround is to use a version of Gradle < 7. We'll make sure to post an update when the fix is released.

**Comment by inktomi** [comment]
As a workaround, in our bash build scripts, we've added the following when we want to use Gradle 7.0 for most of the build, but have to use 6.8 for the App Distribution jobs. It's messy, but works. 

```bash
# WORKAROUND: Intentionally downgrade Gradle to work around App Distribution being incompatible with Gradle 7
echo "distributionUrl=https\://services.gradle.org/distributions/gradle-6.8.3-bin.zip" >> gradle/wrapper/gradle-wrapper.properties

# Run this command after the script exits to revert the previous operation to prevent any propagation or reuse of 6.8.3
trap "git checkout -- gradle/wrapper/gradle-wrapper.properties" EXIT

# END WORKAROUND
```

**Comment by CristianGM** [comment]
2.1.1 fixed the issue for me

**Comment by leinardi** [comment]
> 2.1.1 fixed the issue for me

Can confirm that this works fine with AGP 4.1.3 and Gradle 7.0 :+1:

**Comment by davidbilik** [comment]
Yep, fixed with 2.1.1. Closing the issue.

**Comment by tagboola** [comment]
Great! Version 2.1.1 of the App Distribution Gradle plugin was released yesterday and should fix the issue.  Thanks for confirming!
